contributions/catalog/controllers/contribute.py

```diff
@@ -212,21 +219,140 @@ def contribution_edit(contribution_id):
+                for idx, review_entry in enumerate(review_list):
+                    if review_entry["reviewerId"] == reviewer_id:
+                        review_loc = idx
+                # add new elements(review_json)
+                currenttime = datetime.datetime.now()
+                currenttime = currenttime.strftime("%Y/%m/%dT%H:%M:%S")
+                contribution["review"][review_loc]["lastUpdated"] = currenttime
+                contribution["review"][review_loc]["reviewerComment"] = contribution_comment
+                contribution["review"][review_loc]["reviewerId"] = reviewer_id
+            else:
+                # add new elements
+                review_json = {'review':[{'lastUpdated': currenttime, 'reviewerComment': contribution_comment,
+                                'reviewerId': reviewer_id}]}
+                contribution.update(review_json)
+            contribution["status"] = contribution_status
+            json_contribution = json.dumps(contribution, indent=4)
+            response, s = put_contribution(json_contribution, contribution_id)
+
+            if response:
+                if "name" in session:
+                    return render_template('contribute/submitted.html', user=session["name"],
+                                           token=session['oauth_token']['access_token'])
+                else:
+                    return render_template('contribute/submitted.html')
+            elif not response:
+                if "name" in session:
+                    return render_template('contribute/error.html', user=session["name"],
+                                           token=session['oauth_token']['access_token'], error_msg=s)
+                else:
+                    return render_template('contribute/error.html', error_msg=s)
+    else:
+        if review_flag is not None:
+            if review_flag.lower() == 'true':
+                is_review = True
+
+        the_json_res = get_contribution(contribution_id)
+
+        # check if the user is in contribution's admins
+        if username in the_json_res["contributionAdmins"]:
+            is_editable = True
+
+        # check if the user is a reviewer
+        is_reviewer = adminutil.check_if_reviewer(username, headers)
+
+        # get capability list to create required capability list
+        required_capability_list = requestutil.request_required_capability_list(headers)
+
+        if is_reviewer and is_review:
+            # need to add review information to html so it can pre render if needed
+            reviewer_id = adminutil.get_reviewer_id(headers, username)
+            if "review" in the_json_res.keys():
+                review_exist = False
+                review_list = the_json_res["review"]
+                for idx, review_entry in enumerate(review_list):
+                    if review_entry["reviewerId"] == reviewer_id:
+                        review_loc = idx
+                        review_exist = True
```

Comment:
Do we need a break statement here?

Reply:
Or is it possible to move 
```
review_json = {'review': review_list[review_loc]}
the_json_res.update(review_json)
```
inside the `if review_entry["reviewerId"] == reviewer_id` condition?

Reply:
modified based on your suggestion

Reply:
Thanks.